internal/weldr/api.go

```diff
@@ -3123,35 +3131,42 @@ func (api *API) fetchPackageList(distroName string) (rpmmd.PackageList, error) {
 	return packages, err
 }
 
-// Returns all configured repositories (base, depending on the image type + sources) as rpmmd.RepoConfig
+// Returns only user-defined repositories, which should be used only for
+// payload package sets.
+func (api *API) payloadRepositories(distroName string) []rpmmd.RepoConfig {
+	payloadRepos := make([]rpmmd.RepoConfig, 0, len(api.store.GetAllDistroSources(distroName)))
+	for id, source := range api.store.GetAllDistroSources(distroName) {
```

Comment:
A bit wasteful to be loading sources from the store just to get their length and then loading them again to iterate through.

Reply:
🙄  /me goes fixing it...

Reply:
fixed.